src/lib.rs

```diff
@@ -53,3 +52,37 @@ impl Behaviour {
         }
     }
 }
+
+pub async fn load_secret_key_from_file(path: impl AsRef<Path> + Debug) -> Result<SecretKey> {
+    let bytes = fs::read(&path)
+        .await
+        .context(format!("No secret file at {:?}", path))?;
```

Comment:
Have a look at anyhow's docs regarding `.context` and `with_context`: https://docs.rs/anyhow/1.0.41/anyhow/trait.Context.html#required-methods

Then, explain why you are using `context` here and not `with_context`.